src/overlays/actors/ovl_En_Rg/z_en_rg.c

```diff
@@ -378,13 +377,18 @@ s32 func_80BF43FC(EnRg* this) {
     f32 phi_f20 = 0.0f;
     s32 temp_s7 = ENRG_GET_7F80(&this->actor);
     s32 phi_s4 = -1;
-    s16 phi_s6 = 0;
     s32 temp_s5 = this->unk_344;
+    s16 phi_s6 = 0;
     s32 phi_s0 = D_80BF57E4[this->unk_344][temp_s7];
 
     while (true) {
         SubS_CopyPointFromPathCheckBounds(this->path, phi_s0 - 1, &sp9C);
         SubS_CopyPointFromPathCheckBounds(this->path, phi_s0 + 1, &sp90);
+        if (1) {}
+        if (1) {}
+        if (1) {}
+        if (1) {}
+        if (1) {}
```

Comment:
Each one of these `if (1)`'s changes around the save registers, and with the stack ordering at the top, the permuter found this meme of a sweet spot. So with the current solution, yes all are required. Obviously not ideal...

Reply:
Been looking at this, and was able to remove the `if(1)` conga line with a `do while` loop, instead of breaking out of an infinite loop.

```c
    do {
        SubS_CopyPointFromPathCheckBounds(this->path, phi_s0 - 1, &sp9C);
        SubS_CopyPointFromPathCheckBounds(this->path, phi_s0 + 1, &sp90);
        if (Math3D_PointDistToLine2D(this->actor.world.pos.x, this->actor.world.pos.z, sp9C.x, sp9C.z, sp90.x, sp90.z,
                                     &sp8C, &sp88, &sp84) &&
            (!phi_s6 || ((phi_s4 + 1) == phi_s0) || (sp84 < phi_f20))) {
            phi_s6 = 1;
            phi_f20 = sp84;
            phi_s4 = phi_s0;
        }
        phi_s0++;
    }  while((temp_s5 != 18) && (phi_s0 < D_80BF57E4[temp_s5 + 1][temp_s7]));
```

Reply:
Nice Find!